### ISSUE TITLE
Trainig error

### ISSUE THREAD

**Issue by NeuralCoder3** [meipuru344/sepconv#1]
When I train the pre-trained model with my own images with this code,
it happens that after some time the weird artifacts show up and the contrast increases.
This is independent of the trained image, the optimizer or loss used.
The loss increases with time, so it is unlikely that it is caused by exploding gradients.
Do you have any tips? Have you trained a network with our code on real images?

PS: Thank you for the code. I was trying it my self but it didn't quite work out.  So I was very grateful when I found your code.

**Comment by l2009312042** [comment]
yes ，i have the  same issues ，any ideas？ the loss increase after some iteration，the predicte pic is very bad although the first predict by pre-trained model is ok 



this is the log
Loss: 0.018768  Iterations:   99/1562  TrainExecTime: 47.9  ValLoss:0.050459  ValPSNR: 22.1817  ValEvalTime: 2.62 LearningRate: 0.001000
 Loss: 5.605599  Iterations:  199/1562  TrainExecTime: 48.0  ValLoss:185.799081  ValPSNR: -56.2441  ValEvalTime: 2.54 LearningRate: 0.001000
 Loss: inf  Iterations:  299/1562  TrainExecTime: 47.1  ValLoss:42954.060937  ValPSNR: -91.4158  ValEvalTime: 2.62 LearningRate: 0.001000
 Loss: 3689618865245.907715  Iterations:  399/1562  TrainExecTime: 46.9  ValLoss:85205327858892.796875  ValPSNR: -194.6913  ValEvalTime: 2.57 LearningRate: 0.001000
 Loss: 704643011230262493184.000000  Iterations:  499/1562  TrainExecTime: 46.8  ValLoss:11784975425921760624640.000000  ValPSNR: -284.5473  ValEvalTime: 2.53 LearningRate: 0.001000

**Comment by l2009312042** [comment]
i got it  

	for(int imageDepthD = 0; imageDepthD < 3 ; imageDepthD += 1){
				for(int kernelDepthij = 0; kernelDepthij < 51 ; kernelDepthij += 1){
				singleOutH +=
				VALUE_4(vertical, intSample, kernelDepthij, intY, intX) *
				VALUE_4(input, intSample, imageDepthD, intY + intDepthIJ, intX + kernelDepthij) *
				VALUE_4(gradOut, intSample, imageDepthD, intY, intX);
				singleOutV +=
				VALUE_4(horizontal, intSample, kernelDepthij, intY, intX) *
				VALUE_4(input, intSample, imageDepthD, intY + kernelDepthij, intX + intDepthIJ) *
				VALUE_4(gradOut, intSample, imageDepthD, intY, intX);
				}
			}
 intY + intDepthIJ, intX + kernelDepthij  should be  intY+ kernelDepthij , intX + intDepthIJ

and  intY + kernelDepthij, intX + intDepthIJ should be  intY + kernelDepthij, intX + intDepthIJ
